notifiers/slack/slack.go

```diff
@@ -0,0 +1,197 @@
+
+	"github.com/dustin/go-humanize"
+	"github.com/mitchellh/mapstructure"
+	slackApi "github.com/nlopes/slack"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	// ErrNoToken will be used when there is no token defined
+	ErrNoToken = errors.New("slack token is required")
+)
+
+const (
+	// AuthorName Slack will use while sending the name
+	AuthorName = "Finala Notifier"
+	// MessageColor color to use
+	MessageColor = "#2EB67D"
+)
+
+// NewManager returns the notifier
+func NewManager() notifierCommon.Notifier {
+	return &Manager{}
+}
+
+// LoadConfig maps a generic notifier config (map[string]interface{}) to a concrete type
+func (sm *Manager) LoadConfig(notifierConfig notifierCommon.NotifierConfig) (err error) {
+	newConfig := Config{}
+	if err = mapstructure.Decode(notifierConfig, &newConfig); err != nil {
+		return err
+	}
+
+	// Set the token
+	sm.config.Token = newConfig.Token
+
+	if newConfig.DefaultChannels != nil {
+		sm.config.DefaultChannels = newConfig.DefaultChannels
+	}
+
+	if newConfig.NotifyByTags != nil {
+		sm.config.NotifyByTags = newConfig.NotifyByTags
+	}
+
+	// Validate the slack configuration has a token configured
+	if sm.config.Token == "" {
+		return ErrNoToken
+	}
+
+	// Initialize slack client
+	sm.client = slackApi.New(sm.config.Token)
+
+	return nil
+}
+
+// GetNotifyByTags will get the all the fields of notify_by_tags from slack configuration
+func (sm *Manager) GetNotifyByTags(notifierConfig common.ConfigByName) map[string]notifierCommon.NotifyByTag {
+	notifyByTags := map[string]notifierCommon.NotifyByTag{}
+	if sm.config.NotifyByTags != nil {
+		notifyByTags = sm.config.NotifyByTags
+	}
+	return notifyByTags
+}
+
+// prepareAttachmentFields will prepare all the Attachment and all the fields
+func (sm *Manager) prepareAttachment(message common.NotifierReport, tags []string) []slackApi.Attachment {
+	slackAttachmentFields := []slackApi.Attachment{
+		{
+			Color:      MessageColor,
+			AuthorName: AuthorName,
+			Pretext: fmt.Sprintf("Here is the <%s|Cost report> for your notification group: %s filtered by: %s",
+				message.UIAddr,
+				message.GroupName,
+				strings.Join(tags, " AND ")),
+		}}
+	for _, executionData := range message.ExecutionSummaryData {
+		// If the total spent is 0 we don't want to show it in Slack
+		if executionData.ResourceCount == 0 || executionData.TotalSpent == 0 {
+			continue
+		}
+		slackAttachmentFields = append(slackAttachmentFields, slackApi.Attachment{
+			Color: MessageColor,
+			Fields: []slackApi.AttachmentField{
+				{
+					Title: strings.ToUpper(executionData.ResourceName),
+					Value: fmt.Sprintf("Potential Saving: $%s", humanize.Commaf(executionData.TotalSpent)),
+					Short: false,
+				},
+			},
+		})
+	}
+	return slackAttachmentFields
+}
+
+// Send all slack Notifications to users and channels
+func (sm *Manager) Send(message notifierCommon.NotifierReport) {
+	message.Log.Debug("Updating all current slack users list")
+	err := sm.updateUsers()
```

Comment:
You are getting all the users every time that you send a message? 
This is all the users in slack?  
Why are you not calling this function when you initialize